### ISSUE TITLE
re-fire the same asyncAction that failed 

### ISSUE THREAD

**Issue by NirLud** [welldone-software/redux-toolbelt#42]
Hi, 
First of all, Thank you for this great package!

Now, My problem is: I need to re-fire the same asyncAction that failed (with the same payload, after changing the local storage accessToken value).
How can i do it with out adding it as a metaData to each action? 

*userFireAccessToken is just a selector for the refreshed token.

`
  export const onUnauthorizedResponse = (action$, state$) =>
  action$.pipe(
    switchMap(({type, payload}) => {
      if (type.endsWith('ASYNC_FAILURE') && payload?.response?.status === 401) {
        if(userFireAccessToken(state$.value) && localStorage.getItem('accessToken') !== userFireAccessToken(state$.value)){
          localStorage.setItem('accessToken', userFireAccessToken(state$.value));
          return [
            /// HERE I WANT TO REFIRE THE LAST ACTION (THE ONE THAT FAILD) WITH THE SAME PAYLOAD 
          ];
        }
      }
      return [];
    }),
  );
`

**Comment by gioragutt** [comment]
Word of advice - wrap you code example in triple backticks
> \```javascript
> your code here
> \```

**Comment by gioragutt** [comment]
Will this work?

```javascript
export const onUnauthorizedResponse = (action$, state$) =>
  action$.pipe(
    switchMap(action => {
      // don't destruct immediately, get the full action, and return it later?
      const {type, payload} = action
      if (type.endsWith('ASYNC_FAILURE') && payload?.response?.status === 401) {
        if(userFireAccessToken(state$.value) && localStorage.getItem('accessToken') !== userFireAccessToken(state$.value)){
          localStorage.setItem('accessToken', userFireAccessToken(state$.value));
          return [action];
        }
      }
      return [];
    }),
  );
```

**Comment by NirLud** [comment]
No... you are trying to fire an action of type ASYNC_FAILURE with its payload.
I want to fire the action how fires this failure, The ASYNC_REQUEST.
